Sources/SQL/SQLError.swift

```diff
@@ -0,0 +1,84 @@
+public protocol SQLError: Debuggable {
+  var type: SQLErrorType {get}
+  var reason: String {get}
+  var sourceLocation: SourceLocation? {get}
+  var stackTrace: [String] {get}
+  var identifier: String {get}
+}
+
+extension SQLError {
+  public var identifier: String {
+    return "SQLError(\(self.type.description))"
```

Comment:
The identifier should be a dot-separated identifier, think iOS app bundle ID. Something like:

```
sql.intern
```